### PR TITLE
Improved validation for adding new schemas

### LINKED ISSUE #470: [Enhancement]: Add Structural Validation for Required Field & Object Types Enum
### Description of Enhancement :

As flagged in #457, we had a transaction schema committed for Issuer Authorized Share Adjustment that had the wrong required field name AND an object_type that was not in the object type enum. Either the test failed to trip the GitHub action or our test doesn't cover either situation. I know it doesn't cover checks that fields marked as required actually exist, but I am surprised the missing ObjectType didn't cause a validation error somewhere. 

I propose adding a validation test for BOTH in the validate schema tooling.  

### Why is this Needed?

We didn't catch a schema with two issues making it to a release. 

### Anything else we need to know?

Nope

### PR DESCRIPTION
**Pull request by jacobyavis**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR improves our automated validation to ensure schema integrity against developer errors

#### Which issue(s) this PR fixes:
Fixes #470 

Now, if a developer adds a new Object Type, the validator will force them to add it to the enum, and complete it with a sample as well

This will prevent errors [like we saw in this commit](https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/commit/d48b5fd122a2630b60f18dc7dd820eb8e1e4d49c)

### CODE REVIEW

**Comment by jacobyavis** [comment]
## Sample outputs

### I just added a new schema which references object_type TX_NEW_TRANSACTION_CODE
```
npm run schema:validate-example-ocf-files

> schema:validate-example-ocf-files
> node --loader ts-node/esm --no-warnings --experimental-json-modules ./utils/validate.mjs validate-ocf-directory -p ./samples -t

::error::               OCF Validation failed due to error: Encountered object_type: TX_NEW_TRANSACTION_CODE in a schema but this type does not exist in ObjectType.schema.json
```

### Okay now I've added it to ObjectType
```
npm run schema:validate-all-objects-have-samples

> schema:validate-all-objects-have-samples
> node --loader ts-node/esm --no-warnings --experimental-json-modules ./utils/validate.mjs validate-all-objects-have-samples -t

::error::ObjectType enum value TX_NEW_TRANSACTION_CODE does not appear in the sample OCF files
```

### Okay now I've added a sample
```
npm run schema:validate-all-objects-have-samples
 
> schema:validate-all-objects-have-samples
> node --loader ts-node/esm --no-warnings --experimental-json-modules ./utils/validate.mjs validate-all-objects-have-samples -t
 
 
        ** ALL OBJECT TYPES HAVE SAMPLES **
```

**Comment by jacobyavis** [comment]
Getting a failing check due to recently-merged PR #
```
OCF Validation failed due to error: Encountered object_type: FINANCING in a schema but this type does not exist in ObjectType.schema.json
```
I commented on that PR and raised issue #492 for follow-up

File: schema/objects/StockPlan.schema.json [hunk 1]
```diff
@@ -41,9 +41,9 @@
     "stock_class_ids": {
       "description": "Identifiers of StockClass objects this plan is composed of",
       "type": "array",
+      "minItems": 1,
       "items": {
-        "type": "string",
-        "minItems": 1
```
Comment (jacobyavis) [inline schema/objects/StockPlan.schema.json:46]: The validator was showing a warning about these two occurrences of minItems that were in the wrong spot
```
strict mode: missing type "array" for keyword "minItems" at "https://raw.githubusercontent.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/main/schema/objects/StockPlan.schema.json#/properties/stock_class_ids/items" (strictTypes)

**Review by pjohnmeyer — APPROVED** [review]
LGTM

**Pull request merged**